static-site/src/sections/individual-group-page.jsx

```diff
@@ -51,7 +40,8 @@ class Index extends React.Component {
         sourceInfo,
         groupName,
         datasets: this.createDatasetListing(availableData.datasets, groupName),
-        narratives: this.createDatasetListing(availableData.narratives, groupName)
+        narratives: this.createDatasetListing(availableData.narratives, groupName),
+        missingDataset: this.props["*"]
```

Comment:
Currently this `setState` call happens c. 1000ms (on my machine) after the page actually loads, however the prop is available immediately. (Note that the prop can be assigned to state within the constructor if you wish to rename it, which I agree is sensible.) 

Given that the load time is already affected by the dataset existence check (server-side), we should try to render a useful UI as soon as possible. If we know a dataset is missing, we could display a "something's gone wrong, fetching more information" banner straight away. Furthermore, if we were to enforce all banners to be the same height, then we could simply update this banner with a more informative message as more data arrives. 

There are plenty of options of how exactly to do this, and the first priority should be clarity, with responsiveness (speed) second.